sdk/src/main/scala/com/horizen/websocket/server/WebSocketServerEndpoint.scala

```diff
@@ -0,0 +1,112 @@
+package com.horizen.websocket.server
+
+import java.util
+
+import akka.actor.ActorRef
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.google.inject.Inject
+import javax.websocket.{OnMessage, Session}
+import javax.websocket.server.ServerEndpoint
+
+case object GET_SINGLE_BLOCK_REQUEST_TYPE extends RequestType(0)
+case object GET_NEW_BLOCK_HASHES_REQUEST_TYPE extends RequestType(2)
+case object GET_MEMPOOL_TXS extends RequestType(4)
+case object GET_RAW_MEMPOOL extends RequestType(5)
+
+@ServerEndpoint("/")
+class WebSocketServerEndpoint() {
+  private val mapper = new ObjectMapper().registerModule(DefaultScalaModule)
+
+  @Inject val bean: ActorRef = null;
+
+  @OnMessage
+  def onMessageReceived(session: Session, message: String): Unit = {
+    if (!WebSocketServerEndpoint.sessions.contains(session)) {
+      WebSocketServerEndpoint.addSession(session);
+    }
+    try {
+      val json = mapper.readTree(message)
+      if (json.has("msgType")) {
+        json.get("msgType").asInt() match {
+          case 1 => // Event
+            processRequest(json, session)
+          case 2 => // Error
+            processError(json, session)
+          case msgType =>
+            processError(json, session)
+            System.out.println("Unknown message received with type = " + msgType)
+        }
+      } else {
+        processError(json, session)
+        System.out.println("Unknown message received!")
+      }
+    } catch {
+      case ex: Throwable =>
+        processError( mapper.createObjectNode(), session)
+        System.out.println("On receive message processing exception occurred = " + ex.getMessage)
+    }
+  }
+
+  def processRequest(json: JsonNode,session: Session): Unit = {
+    val sidechainNodeChannelImpl: SidechainNodeChannelImpl = new SidechainNodeChannelImpl(session)
```

Comment:
define SidechainNodeChannelImpl once during `WebSocketServerEndpoint` creation. And as was said above, no need to pass `session`, only nodeviewholderref.